### ISSUE TITLE
Crash caused by dungeoncrawl

### ISSUE THREAD

**Issue by SquirrelCraft** [SquirrelCraft/SquirrelPAK-MC16#2]
During our server testing we had a crash. Looks like dungeoncrawl crashed while generating new chunks

```
Time: 8/6/21, 10:01 PM
Description: Exception generating new chunk

java.lang.IllegalArgumentException: bound must be positive
	at java.util.Random.nextInt(Random.java:388) ~[?:?] {}
	at xiroc.dungeoncrawl.dungeon.generator.layer.NewLayerGenerator.generateLayer(NewLayerGenerator.java:109) ~[dungeoncrawl:2.3.0] {re:classloading}
	at xiroc.dungeoncrawl.dungeon.generator.DefaultDungeonGenerator.generateLayer(DefaultDungeonGenerator.java:62) ~[dungeoncrawl:2.3.0] {re:classloading}
	at xiroc.dungeoncrawl.dungeon.DungeonBuilder.generateLayout(DungeonBuilder.java:144) ~[dungeoncrawl:2.3.0] {re:classloading}
	at xiroc.dungeoncrawl.dungeon.DungeonBuilder.build(DungeonBuilder.java:105) ~[dungeoncrawl:2.3.0] {re:classloading}
	at xiroc.dungeoncrawl.dungeon.Dungeon$Start.generatePieces(Dungeon.java:141) ~[dungeoncrawl:2.3.0] {re:classloading}
	at xiroc.dungeoncrawl.dungeon.Dungeon$Start.func_230364_a_(Dungeon.java:130) ~[dungeoncrawl:2.3.0] {re:classloading}
	at net.minecraft.world.gen.feature.structure.Structure.func_242785_a(Structure.java:224) ~[?:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:performant.mixins.json:world.StructureMixin,pl:mixin:A}
	at net.minecraft.world.gen.feature.StructureFeature.func_242771_a(SourceFile:35) ~[?:?] {re:classloading}
	at net.minecraft.world.gen.ChunkGenerator.func_242705_a(SourceFile:278) ~[?:?] {re:mixin,re:classloading,pl:mixin:APP:betterstrongholds.mixins.json:VanillaStrongholdSpawnMixin,pl:mixin:A}
	at net.minecraft.world.gen.ChunkGenerator.func_242707_a(SourceFile:267) ~[?:?] {re:mixin,re:classloading,pl:mixin:APP:betterstrongholds.mixins.json:VanillaStrongholdSpawnMixin,pl:mixin:A}
	at net.minecraft.world.chunk.ChunkStatus.func_222586_d(ChunkStatus.java:41) ~[?:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:byg.mixins.json:common.world.ChunkStatusMixin,pl:mixin:A}
	at net.minecraft.world.chunk.ChunkStatus.func_223198_a(ChunkStatus.java:198) ~[?:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:byg.mixins.json:common.world.ChunkStatusMixin,pl:mixin:A}
	at net.minecraft.world.server.ChunkManager.lambda$null$18(ChunkManager.java:524) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:APP:ftbchunks-common.mixins.json:ChunkMapMixin,pl:mixin:APP:performant.mixins.json:world.chunk.ChunkManagerMixin,pl:mixin:APP:performant.mixins.json:world.chunk.ChunkManagerUpdatePlayerPosMixin,pl:mixin:APP:performant.mixins.json:world.chunk.ChunkManagerChunksAccessor,pl:mixin:A,pl:runtimedistcleaner:A}
	at com.mojang.datafixers.util.Either$Left.map(Either.java:38) ~[?:?] {re:classloading}
	at net.minecraft.world.server.ChunkManager.lambda$scheduleChunkGeneration$20(ChunkManager.java:522) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:APP:ftbchunks-common.mixins.json:ChunkMapMixin,pl:mixin:APP:performant.mixins.json:world.chunk.ChunkManagerMixin,pl:mixin:APP:performant.mixins.json:world.chunk.ChunkManagerUpdatePlayerPosMixin,pl:mixin:APP:performant.mixins.json:world.chunk.ChunkManagerChunksAccessor,pl:mixin:A,pl:runtimedistcleaner:A}
	at java.util.concurrent.CompletableFuture$UniCompose.tryFire(CompletableFuture.java:1072) ~[?:?] {}
	at java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:478) ~[?:?] {}
	at net.minecraft.world.chunk.ChunkTaskPriorityQueueSorter.func_219083_b(SourceFile:58) ~[?:?] {re:classloading}
	at net.minecraft.util.concurrent.DelegatedTaskExecutor.func_213148_e(SourceFile:94) ~[?:?] {re:classloading}
	at net.minecraft.util.concurrent.DelegatedTaskExecutor.func_213145_a(SourceFile:137) ~[?:?] {re:classloading}
	at net.minecraft.util.concurrent.DelegatedTaskExecutor.run(SourceFile:105) ~[?:?] {re:classloading}
	at java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1426) ~[?:?] {}
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290) ~[?:?] {}
	at java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020) ~[?:?] {}
	at java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656) ~[?:?] {}
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594) ~[?:?] {}
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183) ~[?:?] {}
```

**Comment by SquirrelCraft** [comment]
[crash-2021-08-06_22.01.38-server.txt](https://github.com/SquirrelCraft/SquirrelPAK-MC16/files/6948478/crash-2021-08-06_22.01.38-server.txt)

**Comment by SquirrelCraft** [comment]
Full crash report below
[crash-2021-08-06_22.01.38-server.txt](https://github.com/SquirrelCraft/SquirrelPAK-MC16/files/6948485/crash-2021-08-06_22.01.38-server.txt)

**Comment by SquirrelCraft** [comment]
Was able to load up the mod pack and in single player mode was able to get it to crash. I will try only that mod and use the same seed later 

To get it to crash
Mod Pack MC16 v0.8.2
Seed: -1
Teleport to x=-1005, y=104, z=2363]]

**Comment by SquirrelCraft** [comment]
I tested again using only dungeoncrawl, default config and was able to reproduce the crash using seed -1 and going to the same coord dungeoncrawl Teleport to x=-1005, y=104, z=2363

**Comment by SquirrelCraft** [comment]
submitted an issue on the mod makers issue tracker.

**Comment by SquirrelCraft** [comment]
Looks like the mod author has uploaded a fix, I will test this soon

**Comment by SquirrelCraft** [comment]
Fixed Beta 8.3
